ksql-functional-tests/src/test/java/io/confluent/ksql/test/PlannedTestsUpToDateTest.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.test;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.confluent.ksql.execution.json.PlanJsonMapper;
+import io.confluent.ksql.test.planned.TestCasePlan;
+import io.confluent.ksql.test.planned.TestCasePlanLoader;
+import io.confluent.ksql.test.planned.PlannedTestLoader;
+import io.confluent.ksql.test.planned.PlannedTestUtils;
+import io.confluent.ksql.test.tools.TestCase;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+/**
+ * Test that ensures that each QTT test case that should be tested from a physical
+ * plan has the latest physical plan written to the local filesystem.
+ */
+@RunWith(Parameterized.class)
+public class PlannedTestsUpToDateTest {
+  private static final ObjectMapper MAPPER = PlanJsonMapper.create();
+
+  private final TestCase testCase;
+
+  @Parameterized.Parameters(name = "{0}")
+  public static Collection<Object[]> data() {
+    return QueryTranslationTest.findTestCases()
+        .filter(PlannedTestUtils::isPlannedTestCase)
+        .map(testCase -> new Object[]{testCase.getName(), testCase})
+        .collect(Collectors.toList());
+  }
+
+  /**
+   * Test to check for qtt cases that require a new plan to be generated/persisted
+   *
+   * @param name unused - included just so the test has a name
+   * @param testCase test case to check for requiring plan generation
+   */
+  public PlannedTestsUpToDateTest(final String name, final TestCase testCase) {
+    this.testCase = Objects.requireNonNull(testCase);
+  }
+
+  @Test
+  public void shouldHaveLatestPlans() {
```

Comment:
This is failing the build right now since the new plan schema and topology files weren't added.